browser_history/generic.py

```diff
@@ -249,62 +255,56 @@ class Outputs:
-        respective variables and formatting fields
-    * **format_map**: Dictionary which maps output formats to their respective functions
-
+    :param fetch_type: string argument to select history output or bookmarks output
     """
+    histories: HistoryVar = [] #: List of tuples of Timestamp & URL
+    bookmarks: BookmarkVar = []
+    """List of tuples of Timestamp , URL , Title , Folder."""
+    field_map = {
+        "history": {"var": histories, "fields": ("Timestamp", "URL")},
+        "bookmarks": {
+            "var": bookmarks,
+            "fields": ("Timestamp", "URL", "Title", "Folder"),
+        },
+    }
+    """Dictionary which maps fetch_type to the respective variables and formatting fields."""
 
     def __init__(self, fetch_type):
-        self.histories = []
-        self.bookmarks = []
-
         self.fetch_type = fetch_type
 
-        self.field_map = {
-            "history": {"var": self.histories, "fields": ("Timestamp", "URL")},
-            "bookmarks": {
-                "var": self.bookmarks,
-                "fields": ("Timestamp", "URL", "Title", "Folder"),
-            },
-        }
-
-        self.format_map = {
-            "csv": self.to_csv,
-            "json": self.to_json,
-            "jsonl": lambda: self.to_json(json_lines=True),
-        }
-
-    def sort_domain(self):
+    def sort_domain(self) -> Union[HistoryVar, BookmarkVar]:
         """
         Returns the history/bookamarks sorted according to the domain-name.
 
-        :rtype: dict()
-                :type dict.key: str
-                :type dict.value: list(tuple(:py:class:`datetime.datetime`, str))
-                or
-                dict()
-                :type dict.key: str
-                :type dict.value: list(tuple(:py:class:`datetime.datetime`, str, str, str))
+        Examples:
+
+        >>> from datetime import datetime
+        ... from browser_history import generic
+        ... entries = [
+        ...     [datetime(2020, 1, 1), 'https://google.com'],
+        ...     [datetime(2020, 1, 1), 'https://google.com/imghp?hl=EN'],
+        ...     [datetime(2020, 1, 1), 'https://example.com'],
+        ... ]
+        ... obj = generic.Outputs('history')
+        ... obj.histories = entries
+        ... obj.sort_domain()
+        defaultdict(<class 'list'>, {
+            'example.com': [[datetime.datetime(2020, 1, 1, 0, 0), 'https://example.com']],
+            'google.com': [
+                 [datetime.datetime(2020, 1, 1, 0, 0), 'https://google.com'],
+                 [datetime.datetime(2020, 1, 1, 0, 0), 'https://google.com/imghp?hl=EN']]
+         })
         """
         domain_histories = defaultdict(list)
         for entry in self.field_map[self.fetch_type]["var"]:
             domain_histories[urlparse(entry[1]).netloc].append(entry)
         return domain_histories
 
-    def formatted(self, output_format="csv"):
+    def formatted(self, output_format: str="csv") -> str:
```

Comment:
Another small thing:
```suggestion
    def formatted(self, output_format: str = "csv") -> str:
```
(this was suggested by the [black](https://github.com/psf/black/) formatter)